src/DRICH_geo.cpp

```diff
@@ -132,6 +139,17 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
     gasvolVis = vesselVis = desc.invisible();
   };
 
+  // sensor size rescaling for large sensor (catching all photons)
+  double sensorRescale = 0;
+  if (debugOpticsMode == 4) sensorRescale = 250;
+  if (sensorRescale > 0) {
+    auto seg = (CartesianGridXY) desc.readout(readoutName).segmentation();
```

Comment:
Avoid C-style casts. In this case the behavior is unspecified when a segmentation is returned that cannot be cast into `CartesianGridXY`.